cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/DurableLogInspectCommand.java

```diff
@@ -0,0 +1,323 @@
+                    ((MergeSegmentOperation) op).getStreamSegmentId(), ((MergeSegmentOperation) op).getStreamSegmentOffset(),
+                    ((MergeSegmentOperation) op).getAttributeUpdates().size());
+        } else if (op instanceof UpdateAttributesOperation) {
+            res = new OperationInspectInfo(op.getSequenceNumber(), op.getClass().getSimpleName(), op.getCacheLength(),
+                    ((UpdateAttributesOperation) op).getStreamSegmentId(), OperationInspectInfo.DEFAULT_ABSENT_VALUE,
+                    OperationInspectInfo.DEFAULT_ABSENT_VALUE);
+        } else if (op instanceof StreamSegmentTruncateOperation) {
+            res = new OperationInspectInfo(op.getSequenceNumber(), op.getClass().getSimpleName(), op.getCacheLength(),
+                    ((StreamSegmentTruncateOperation) op).getStreamSegmentId(), ((StreamSegmentTruncateOperation) op).getStreamSegmentOffset(),
+                    OperationInspectInfo.DEFAULT_ABSENT_VALUE);
+        } else if (op instanceof DeleteSegmentOperation) {
+            res = new OperationInspectInfo(op.getSequenceNumber(), op.getClass().getSimpleName(), op.getCacheLength(),
+                    ((DeleteSegmentOperation) op).getStreamSegmentId(), ((DeleteSegmentOperation) op).getStreamSegmentOffset(),
+                    OperationInspectInfo.DEFAULT_ABSENT_VALUE);
+        } else if (op instanceof MetadataCheckpointOperation) {
+            res = new OperationInspectInfo(op.getSequenceNumber(), op.getClass().getSimpleName(), op.getCacheLength(),
+                    OperationInspectInfo.DEFAULT_ABSENT_VALUE, OperationInspectInfo.DEFAULT_ABSENT_VALUE, OperationInspectInfo.DEFAULT_ABSENT_VALUE);
+        } else if (op instanceof StorageMetadataCheckpointOperation) {
+            res = new OperationInspectInfo(op.getSequenceNumber(), op.getClass().getSimpleName(), op.getCacheLength(),
+                    OperationInspectInfo.DEFAULT_ABSENT_VALUE, OperationInspectInfo.DEFAULT_ABSENT_VALUE, OperationInspectInfo.DEFAULT_ABSENT_VALUE);
+        } else if (op instanceof StreamSegmentMapOperation) {
+            res = new OperationInspectInfo(op.getSequenceNumber(), op.getClass().getSimpleName(), op.getCacheLength(),
+                    ((StreamSegmentMapOperation) op).getStreamSegmentId(), OperationInspectInfo.DEFAULT_ABSENT_VALUE, OperationInspectInfo.DEFAULT_ABSENT_VALUE);
+        }
+        return res;
+    }
+
+    private int filterResult(Predicate<OperationInspectInfo> predicate, int containerId, DebugDurableDataLogWrapper originalDataLog) throws Exception {
+        AtomicInteger res = new AtomicInteger();
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yy.HH-mm-ss");
+        Date date = new Date();
+        @Cleanup
+        FileWriter writer = new FileWriter(createFileAndDirectory("/tmp/inspect/DurableLogInspectResult" + dateFormat.format(date)));
```

Comment:
Looks like this file consist results of DurableLogInspectResult command, instead of hardcoding this path in code is it better to take from some config file?

Reply:
updated to read from admin-cli.properties files